dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreTextSearch.cs

```diff
@@ -231,8 +230,11 @@ private async IAsyncEnumerable<object> GetResultsAsRecordAsync(IAsyncEnumerable<
 
         await foreach (var result in searchResponse.WithCancellation(cancellationToken).ConfigureAwait(false))
         {
-            yield return result.Record;
-            await Task.Yield();
+            if (result.Record != null)
+            {
+                yield return result.Record;
+                await Task.Yield();
+            }
```

Comment:
if `result.Record is null` do we want to silently proceed?

In general we favour `if (result.Record is not null)`

Reply:
Updated to `is not null`.

None of the connectors today return nulls since they are not really useful, especially in a search context.  This new check is an unfortunate side affect of removing the class constraint.  Since the compiler doesn't know what type TRecord is, it assumes that it might be null, so forces a check on us.